blocks/library/index.js

```diff
@@ -35,6 +35,26 @@ import * as video from './video';
 
 export const registerCoreBlocks = () => {
 	[
+		// FIXME: Temporary fix.
+		//
+		// The Shortcode block declares a catch-all shortcode transform,
+		// meaning it will attempt to intercept pastes and block conversions of
+		// any valid shortcode-like content. Other blocks (e.g. Gallery) may
+		// declare specific shortcode transforms (e.g. `[gallery]`), with which
+		// this block would conflict. Thus, the Shortcode block needs to be
+		// registered as early as possible, so that any other block types'
+		// shortcode transforms can be honoured.
+		//
+		// This isn't a proper solution, as it is at odds with the
+		// specification of shortcode conversion, in the sense that conversion
+		// is explicitly independent of block order. Thus, concurrent parse
+		// rules (i.e. a same text input can yield two different transforms,
+		// like `[gallery] -> { Gallery, Shortcode }`) are unsupported,
+		// yielding non-deterministic results. A proper solution could be to
+		// let the editor (or site owners) determine a default block handler of
+		// unknown shortcodes — see `setUnknownTypeHandlerName`.
+		shortcode,
```

Comment:
How about moving it out of this array to make things extra clear? `registerBlockType( shortcode.name, shortcode.settings );`?

Reply:
Both work for me :) Hopefully it will be fixed soon as described in the comment.

Reply:
Not sure. It feels like all the core blocks should be gathered in a single place, and the current approach of creating an array and looping over it seems more apt. Placing a `registerBlockType` call for Shortcode beside that loop could perhaps make it less visible?